tests/test_pytorch_sparse_dataloader_api.py

```diff
@@ -0,0 +1,249 @@
+    def __init__(self, shape):
+        super(Net, self).__init__()
+        self.flatten = nn.Flatten()
+        self.linear_relu_stack = nn.Sequential(
+            nn.Linear(np.product(shape), 32),
+            nn.ReLU(),
+            nn.Linear(32, 32),
+            nn.ReLU(),
+            nn.Linear(32, NUM_OF_CLASSES),
+            nn.ReLU(),
+        )
+
+    def forward(self, x):
+        x = self.flatten(x)
+        logits = self.linear_relu_stack(x)
+        return logits
+
+
+class TestTileDBSparsePyTorchDataloaderAPI(unittest.TestCase):
+    def test_tiledb_pytorch_sparse_data_api_train_with_multiple_dim_data(self):
+        for input_shape in INPUT_SHAPES:
+            for workers in NUM_OF_WORKERS:
+                with self.subTest():
+                    with tempfile.TemporaryDirectory() as tiledb_uri_x, tempfile.TemporaryDirectory() as tiledb_uri_y:
+
+                        dataset_shape_x = (ROWS, input_shape)
+                        dataset_shape_y = (ROWS,)
+
+                        ingest_in_tiledb_sparse(
+                            uri=tiledb_uri_x,
+                            data=create_sparse_array_one_hot_2d(
+                                dataset_shape_x[0], dataset_shape_x[1]
+                            ),
+                            batch_size=BATCH_SIZE,
+                        )
+                        ingest_in_tiledb(
+                            uri=tiledb_uri_y,
+                            data=np.random.randint(
+                                low=0, high=NUM_OF_CLASSES, size=dataset_shape_y
+                            ),
+                            batch_size=BATCH_SIZE,
+                        )
+
+                        with tiledb.open(tiledb_uri_x) as x, tiledb.open(
+                            tiledb_uri_y
+                        ) as y:
+
+                            tiledb_dataset = PyTorchTileDBSparseDataset(
+                                x_array=x, y_array=y, batch_size=BATCH_SIZE
+                            )
+
+                            self.assertIsInstance(
+                                tiledb_dataset, torch.utils.data.IterableDataset
+                            )
+
+                            train_loader = torch.utils.data.DataLoader(
+                                tiledb_dataset, batch_size=None, num_workers=workers
+                            )
+
+                            # Train network
+                            net = Net(shape=dataset_shape_x[1:])
+                            criterion = torch.nn.CrossEntropyLoss()
+                            optimizer = optim.Adam(
+                                net.parameters(),
+                                lr=0.001,
+                                betas=(0.9, 0.999),
+                                eps=1e-08,
+                            )
+
+                            for epoch in range(
+                                1
+                            ):  # loop over the dataset multiple times
```

Comment:
Having a loop that runs only once looks strange; does this work with more than one epochs?

Btw move the comment just before the loop to prevent `black` for adding this ugly line break.

Reply:
Yeap we have put it there mostly for semantics. There can be and should be more epochs in normal training examples. For testing we use `1`. I would suggest to leave it this way, as it is easily recognised by a user.